ingestion_server/test/unit_tests/test_slack.py

```diff
@@ -57,3 +57,62 @@ def test_message(
         assert data["text"] == expected_summary
         if environment:
             assert data["username"].endswith(environment.upper())
+
+@pytest.mark.parametrize(
+    "log_level, should_log",
+    [
+        ("ERROR", False),
+        ("INFO", False),
+        ("VERBOSE", True),
+        # If no log level set, verbose logging is enabled by default
+        (None, True)
+    ]
+)
+def test_verbose(log_level, should_log, monkeypatch):
+    monkeypatch.setenv("ENVIRONMENT", "staging")
+    monkeypatch.setenv(slack.SLACK_WEBHOOK, "http://fake")
+    if log_level:
+        monkeypatch.setenv(slack.LOG_LEVEL, log_level)
+    with mock.patch("requests.post") as mock_post:
+        slack.verbose("text", "summary")
+        assert mock_post.called == should_log
+
+
+@pytest.mark.parametrize(
+    "log_level, should_log",
+    [
+        ("ERROR", False),
+        ("INFO", True),
+        ("VERBOSE", True),
+        # If no log level set, verbose logging is enabled by default
+        (None, True)
+    ]
+)
+def test_info(log_level, should_log, monkeypatch):
+    monkeypatch.setenv("ENVIRONMENT", "staging")
+    monkeypatch.setenv(slack.SLACK_WEBHOOK, "http://fake")
+    if log_level:
+        monkeypatch.setenv(slack.LOG_LEVEL, log_level)
+    with mock.patch("requests.post") as mock_post:
+        slack.info("text", "summary")
+        assert mock_post.called == should_log
+
+
+@pytest.mark.parametrize(
+    "log_level, should_log",
+    [
+        ("ERROR", True),
+        ("INFO", True),
+        ("VERBOSE", True),
+        # If no log level set, verbose logging is enabled by default
+        (None, True)
+    ]
+)
+def test_error(log_level, should_log, monkeypatch):
+    monkeypatch.setenv("ENVIRONMENT", "staging")
+    monkeypatch.setenv(slack.SLACK_WEBHOOK, "http://fake")
+    if log_level:
+        monkeypatch.setenv(slack.LOG_LEVEL, log_level)
+    with mock.patch("requests.post") as mock_post:
+        slack.error("text", "summary")
+        assert mock_post.called == should_log
```

Comment:
FWIW these tests could be collapsed into a single parametrization:
```suggestion
@pytest.mark.parametrize(
    "log_func, log_level, should_log",
    [
        # Verbose logging
        (slack.verbose, "ERROR", False),
        (slack.verbose, "INFO", False),
        (slack.verbose, "VERBOSE", True),
        # Info logging
        (slack.info, "ERROR", False),
        (slack.info, "INFO", True),
        (slack.info, "VERBOSE", True),
        # Error logging
        (slack.error, "ERROR", True),
        (slack.error, "INFO", True),
        (slack.error, "VERBOSE", True),
        # If no log level set, verbose logging is enabled by default
        (slack.verbose, None, True)
        (slack.info, None, True)
        (slack.error, None, True)
    ]
)
def test_verbose(log_func, log_level, should_log, monkeypatch):
    monkeypatch.setenv("ENVIRONMENT", "staging")
    monkeypatch.setenv(slack.SLACK_WEBHOOK, "http://fake")
    if log_level:
        monkeypatch.setenv(slack.LOG_LEVEL, log_level)
    with mock.patch("requests.post") as mock_post:
        log_func("text", "summary")
        assert mock_post.called == should_log
```

Reply:
That's true -- I was resisting collapsing the tests for multiple functions. But they are extremely related and it certainly is a lot of duplication!

Reply:
I hear you - I think if there were anything different within the tests besides the actual function call, it would make sense to have them separate!